### PR TITLE
[WIP] Add chocolatey folder containing configuration for choco pack and evntually choco push

### PR DESCRIPTION
**Pull request by heynickc**
Adds basic Chocolatey configuration in preparation for running `choco push`.  Running `choco pack` in the `installers\1.6.2\chocolatey\spark-installer-windows` will output a valid .nupkg that can be pushed.  This is re: issue #7

### CODE REVIEW

**Comment by heynickc** [comment]
Current usage, when pushed to Chocolatey community is `choco install spark-installer-windows`.  It will automatically run the 1.6.2 spark installation script.  There is a known bug in chocolatey that causes the stdout of the Powershell script to output as stderr.  This will be fixed in the next release of Chocolatey per this issue:  https://github.com/chocolatey/choco/issues/1016

The script will run successfully and return exit code 0, however it is red in the console.

**Comment by Aaronontheweb** [comment]
@heynickc IMHO, shouldn't include the .nupkg in the commit. And could we adjust the readme to not include any boilerplate, but just the instructions needed to build this package specifically?

**Comment by heynickc** [comment]
@Aaronontheweb ok, I updated per the latest commits.  I completely removed the choco-generated readme for now, since pushing to chocolatey is an internal concern and should only be pushed to our official chocolatey repository.  Also comments in nuspec.  Tagging this as WIP before we push to chocolatey, as we'll need to update VERIFICATION.txt and LICENSE.txt appropriately per your input.

**Comment by ferventcoder** [comment]
Just a note - https://github.com/petabridge/spark-installer-windows/issues/2#issuecomment-289889023

File: installers/1.6.2/chocolatey/spark-installer-windows/tools/chocolateyinstall.ps1 [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿$ErrorActionPreference = 'Stop'; # stop on all errors
+
+$packageName= 'spark-installer-windows' # arbitrary name for the package, used in messages
+$toolsDir   = "$(Split-Path -parent $MyInvocation.MyCommand.Definition)"
+$url        = 'https://raw.githubusercontent.com/petabridge/spark-installer-windows/master/installers/1.6.2/install-spark1.6.2.ps1'
+$fileLocation = Join-Path $toolsDir 'install-spark1.6.2.ps1'
+
+Get-ChocolateyWebFile $packageName $fileLocation $url
```
Comment (ferventcoder) [inline installers/1.6.2/chocolatey/spark-installer-windows/tools/chocolateyinstall.ps1:8]: Why not just put the PowerShell file ***in*** the package?

Comment (ferventcoder) [reply installers/1.6.2/chocolatey/spark-installer-windows/tools/chocolateyinstall.ps1:8]: It's also better to split out this stuff into multiple packages, set proper dependencies, and not just what was done in the script.

File: installers/1.6.2/chocolatey/spark-installer-windows/tools/chocolateyuninstall.ps1 [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+}
+
+
+## OTHER HELPERS
+## https://chocolatey.org/docs/helpers-reference
+#Uninstall-ChocolateyZipPackage $packageName # Only necessary if you did not unpack to package directory - see https://chocolatey.org/docs/helpers-uninstall-chocolatey-zip-package
+#Uninstall-ChocolateyEnvironmentVariable # 0.9.10+ - https://chocolatey.org/docs/helpers-uninstall-chocolatey-environment-variable 
+#Uninstall-BinFile # Only needed if you used Install-BinFile - see https://chocolatey.org/docs/helpers-uninstall-bin-file
+## Remove any shortcuts you added
+
```
Comment (ferventcoder) [inline installers/1.6.2/chocolatey/spark-installer-windows/tools/chocolateyuninstall.ps1:86]: This file needs cleaned up.

File: installers/1.6.2/chocolatey/spark-installer-windows/tools/chocolateybeforemodify.ps1 [hunk 3]
```diff
@@ -0,0 +1,9 @@
+﻿# This runs in 0.9.10+ before upgrade and uninstall.
+# Use this file to do things like stop services prior to upgrade or uninstall.
+# NOTE: It is an anti-pattern to call chocolateyUninstall.ps1 from here. If you
+#  need to uninstall an MSI prior to upgrade, put the functionality in this
+#  file without calling the uninstall script. Make it idempotent in the
+#  uninstall script so that it doesn't fail when it is already uninstalled.
+# NOTE: For upgrades - like the uninstall script, this script always runs from 
+#  the currently installed version, not from the new upgraded package version.
+
```
Comment (ferventcoder) [inline installers/1.6.2/chocolatey/spark-installer-windows/tools/chocolateybeforemodify.ps1:9]: This should be removed.

File: installers/1.6.2/chocolatey/spark-installer-windows/tools/VERIFICATION.txt [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+Note: Include this file if including binaries you have the right to distribute. 
+Otherwise delete. this file.
+
+===DELETE ABOVE THIS LINE AND THIS LINE===
+
+VERIFICATION
+Verification is intended to assist the Chocolatey moderators and community
+in verifying that this package's contents are trustworthy.
+ 
+<Include details of how to verify checksum contents>
```
Comment (ferventcoder) [inline installers/1.6.2/chocolatey/spark-installer-windows/tools/VERIFICATION.txt:11]: Remove if no binaries in the package.

File: installers/1.6.2/chocolatey/spark-installer-windows/tools/LICENSE.txt [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+Note: Include this file if including binaries you have the right to distribute. 
+Otherwise delete. this file.
+
+===DELETE ABOVE THIS LINE AND THIS LINE===
+
+From: <insert applicable license url here>
+
+LICENSE
+
+<Insert License Here>
```
Comment (ferventcoder) [inline installers/1.6.2/chocolatey/spark-installer-windows/tools/LICENSE.txt:11]: Remove if no binaries in the package as per the instructions.

**Review by ferventcoder — COMMENTED** [review]
Left a few notes.

**Comment by heynickc** [comment]
@ferventcoder Thanks so much Rob!  Any insight on a way I can patch the Chocolatey `Start-ChocolateyProcessAsAdmin` helper to prevent this issue? https://github.com/chocolatey/choco/issues/1016

**Comment by ferventcoder** [comment]
It's likely you don't even need to run another script. The idea in the first place was to adapt into three packages and use Chocolatey packaging components there.

**Pull request merged**